components/Activitylist.js

```diff
@@ -40,6 +49,33 @@ export default function Activitylist({
     }
   }
 
+  function handleAddFavoriteButton(id) {
+    const activitySuggestionCard = activityCards.find((card) => card.id === id);
+
+    if (activitySuggestionCard) {
+      setActivityCards(
+        activityCards.map((card) =>
+          card.id === id
+            ? {
+                ...card,
+                userData: card.userData.some((user) => user.userID === userID)
+                  ? card.userData.filter((user) => user.userID !== userID)
+                  : [
+                      ...card.userData,
+                      {
+                        userID: userID,
+                        userName: userName,
+                      },
+                    ],
```

Comment:
I would suggest you only save the userIDs, because if a user changes their username in the future in their profile, all your cards are then outdated and you would need to find every single one and update them. If you store only the reference (id) to the user, you can search and insert the username later on.